src/Sulu/Component/Content/Metadata/Parser/PropertiesXmlParser.php

```diff
@@ -98,6 +99,21 @@ private function loadProperty(\DOMXPath $xpath, \DOMNode $node, &$tags)
         $result['tags'] = $this->loadTags($tags, $xpath, $node);
         $result['params'] = $this->loadParams('x:params/x:param', $xpath, $node);
         $result['meta'] = $this->loadMeta($xpath, $node);
+        $result['types'] = $this->loadTypes($tags, $xpath, $node, $formKey);
+
+        $typeNames = \array_map(function($type) {
+            return $type['name'];
+        }, $result['types']);
+
+        if (!empty($typeNames)) {
+            if (!$result['default-type'] && null !== ($key = \array_key_first($typeNames))) {
+                $result['default-type'] = $typeNames[$key];
+            }
+
+            if (!\in_array($result['default-type'], $typeNames)) {
+                throw new InvalidBlockDefaultTypeException($result['name'], $result['default-type'], $typeNames);
```

Comment:
Shouldn't we create a new one for that? Maybe we can even deprecate all the existing block classes?

Reply:
Or are the current classes still being used if the a `block` tag is written? In that case we wouldn't be able to deprecate them... Would still make a separate class, because this line is definitely not about blocks.